etna/whatson/models.py

```diff
@@ -0,0 +1,621 @@
+    ]
+
+
+class Speaker(Orderable):
+    """
+    This model is used to add speaker information to event pages.
+    """
+
+    page = ParentalKey(
+        "wagtailcore.Page",
+        on_delete=models.CASCADE,
+        related_name="speakers",
+    )
+
+    name = models.CharField(
+        max_length=100,
+        verbose_name=_("name"),
+        blank=False,
+    )
+
+    description = models.CharField(
+        max_length=200,
+        verbose_name=_("description"),
+        blank=True,
+    )
+
+    image = models.ForeignKey(
+        get_image_model_string(),
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+
+    panels = [
+        FieldPanel("name"),
+        FieldPanel("description"),
+        FieldPanel("image"),
+    ]
+
+
+class EventSession(models.Model):
+    """
+    This model is used to add sessions to an event
+    e.g. 28th September @ 9:00, 29th September @ 10:30, 30th September @ 12:00.
+    These will link to the Eventbrite page.
+    """
+
+    page = ParentalKey(
+        "wagtailcore.Page",
+        on_delete=models.CASCADE,
+        related_name="sessions",
+    )
+
+    start = models.DateTimeField(
+        verbose_name=_("starts at"),
+        blank=False,
+    )
+
+    end = models.DateTimeField(
+        verbose_name=_("ends at"),
+        blank=False,
+    )
+
+    panels = [
+        FieldPanel("start"),
+        FieldPanel("end"),
+    ]
+
+    class Meta:
+        verbose_name = _("session")
+        verbose_name_plural = _("sessions")
+        ordering = ["start"]
+
+
+class WhatsOnPage(BasePageWithIntro):
+    """WhatsOnPage
+
+    A page for listing events.
+    """
+
+    @cached_property
+    def events(self):
+        """
+        Returns a queryset of events that are children of this page.
+        """
+        return (
+            EventPage.objects.child_of(self)
+            .live()
+            .public()
+            .order_by("start_date")
+        )
+
+    # DataLayerMixin overrides
+    gtm_content_group = "What's On"
+
+    class Meta:
+        verbose_name = _("What's On page")
+
+    parent_page_types = [
+        "home.HomePage",
+    ]
+    subpage_types = [
+        "whatson.EventPage",
+    ]
+
+    max_count = 1
+
+    content_panels = BasePageWithIntro.content_panels
+
+
+class EventPage(ArticleTagMixin, TopicalPageMixin, BasePageWithIntro):
+    """EventPage
+
+    A page for an event.
+    """
+
+    # Content
+    lead_image = models.ForeignKey(
+        get_image_model_string(),
+        null=True,
+        blank=False,
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+
+    # Event information
+    event_type = models.ForeignKey(
+        EventType,
+        null=True,
+        blank=False,
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+
+    start_date = models.DateTimeField(
+        verbose_name=_("start date"),
+        null=True,
+        blank=True,
```

Comment:
NOTE: There's no need to specify `blank` for non-editable fields, as `blank` has to do with form field behaviour, which is no longer applicable